airflow-core/src/airflow/executors/executor_loader.py

```diff
@@ -211,7 +211,15 @@ def _get_team_executor_configs(cls) -> list[tuple[str | None, list[str]]]:
                 executor_names = team_executor_config.strip("=")
             else:
                 cls.block_use_of_multi_team()
-                team_name, executor_names = team_executor_config.split("=")
+                if conf.getboolean("core", "multi_team", fallback=False):
```

Comment:
Should we remove `cls.block_use_of_multi_team()`? It is using `AIRFLOW__DEV__MULTI_TEAM_MODE` which seems to be another flag?